tests/Orckestra.Composer.Cart.Tests/Services/Order/OrderHistoryViewService_GetOrderDetailViewModelAsync.cs

```diff
@@ -127,5 +129,141 @@ public async Task WHEN_order_is_not_customer_one_SHOULD_return_null()
+                    OrderInfos = new OrderDetailInfoViewModel()
+                    {
+                        IsOrderEditable = true
+                    }
+                });
+
+            //Act
+            var param = new GetCustomerOrderParam
+            {
+                BaseUrl = GetRandom.String(32),
+                CountryCode = GetRandom.String(32),
+                CultureInfo = new CultureInfo("en-US"),
+                Scope = GetRandom.String(32),
+                CustomerId = customerId,
+                OrderNumber = GetRandom.String(32),
+            };
+
+            var result =
+                await orderHistoryViewModelService.GetOrderDetailViewModelAsync(param).ConfigureAwait(false);
+
+            //Assert
+            result.OrderInfos.IsOrderEditable.Should().BeTrue();
+        }
+
+        [Test]
+        [TestCase("New|Pending", new string[] { "Canceled" })]
+        [TestCase("New|Pending", new string[] { "New,Canceled" })]
+        public async Task WHEN_valid_request_EditableShipmentStates_SHOULD_return_FALSE(string editableShipmentStates, string[] cartShipmentStatuses)
+        {
+            //Arrange
+            var customerId = Guid.NewGuid();
+
+            var orderHistoryViewModelService = _container.CreateInstance<OrderHistoryViewService>();
+            _container.GetMock<IOrderRepository>()
+                .Setup(r => r.GetOrderAsync(It.IsAny<GetCustomerOrderParam>()))
+                .ReturnsAsync(new Overture.ServiceModel.Orders.Order
+                {
+                    CustomerId = customerId.ToString()
+                });
+
+            _container.GetMock<IOrderRepository>()
+                .Setup(r => r.GetOrderSettings(It.IsAny<string>()))
+                .ReturnsAsync(new OrderSettings()
+                {
+                    EditableShipmentStates = editableShipmentStates
+
+                });
+
+            var shipmentList = new List<Shipment>();
+            cartShipmentStatuses.ToList().ForEach(
+                item => shipmentList.Add(new Shipment() { Status = item }));
+
+            _container.GetMock<IOrderRepository>()
+                .Setup(r => r.GetOrderAsync(It.IsAny<GetOrderParam>()))
+                .ReturnsAsync(new Overture.ServiceModel.Orders.Order()
+                {
+                    Cart = new Overture.ServiceModel.Orders.Cart
+                    {
+                        Shipments = shipmentList,
+                        Total = 3
+                    },
+                    CustomerId = customerId.ToString()
+                });
+
+            _container.GetMock<IOrderDetailsViewModelFactory>()
+                .Setup(r => r.CreateViewModel(It.IsAny<CreateOrderDetailViewModelParam>()))
+                .Returns(new OrderDetailViewModel()
+                {
+                    OrderInfos = new OrderDetailInfoViewModel()
+                    {
+                        IsOrderEditable = false
+                    }
+                });
```

Comment:
before result.OrderInfos.IsOrderEditable.Should().BeFalse(); you set up here false.
What are you testing here?